Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -0,0 +1,2212 @@
+  else
+    is_d_space_ = false;
+
+  // spectra to fit
+  int start_wi = getProperty("StartWorkspaceIndex");
+  if (isEmpty(start_wi))
+    m_startWorkspaceIndex = 0;
+  else
+    m_startWorkspaceIndex = static_cast<size_t>(start_wi);
+
+  // last spectrum's workspace index, which is included
+  int stop_wi = getProperty("StopWorkspaceIndex");
+  if (isEmpty(stop_wi))
+    m_stopWorkspaceIndex = m_inputMatrixWS->getNumberHistograms() - 1;
+  else {
+    m_stopWorkspaceIndex = static_cast<size_t>(stop_wi);
+    if (m_stopWorkspaceIndex > m_inputMatrixWS->getNumberHistograms() - 1)
+      m_stopWorkspaceIndex = m_inputMatrixWS->getNumberHistograms() - 1;
+  }
+
+  // optimizer, cost function and fitting scheme
+  m_minimizer = getPropertyValue("Minimizer");
+  m_costFunction = getPropertyValue("CostFunction");
+  fit_peaks_from_right_ = getProperty("FitFromRight");
+  constrain_peaks_position_ = getProperty("ConstrainPeakPositions");
+
+  // Peak centers, tolerance and fitting range
+  ProcessInputPeakCenters();
+  // check
+  if (m_numPeaksToFit == 0)
+    throw std::runtime_error("number of peaks to fit is zero.");
+  // about how to estimate the peak width
+  m_peakDSpacePercentage = getProperty("PeakWidthPercent");
+  if (isEmpty(m_peakDSpacePercentage))
+    m_peakDSpacePercentage = -1;
+  else if (m_peakDSpacePercentage < 0)
+    throw std::invalid_argument(
+        "Peak D-spacing percentage cannot be negative!");
+  g_log.debug() << "DeltaD/D = " << m_peakDSpacePercentage << "\n";
+
+  // set up background
+  m_highBackground = getProperty("HighBackground");
+  m_bkgdSimga = getProperty("FindBackgroundSigma");
+
+  // Set up peak and background functions
+  processInputFunctions();
+  // about peak width and other peak parameter estimating method
+  if (is_d_space_ && m_peakDSpacePercentage > 0)
+    peak_width_estimate_approach_ = EstimatePeakWidth::InstrumentResolution;
+  else if (m_peakFunction->name().compare("Gaussian") == 0)
+    peak_width_estimate_approach_ = EstimatePeakWidth::Observation;
+  else
+    peak_width_estimate_approach_ = EstimatePeakWidth::NoEstimation;
+  g_log.debug() << "Process inputs [3] peak type: " << m_peakFunction->name()
+                << ", background type: " << m_bkgdFunction->name() << "\n";
+
+  ProcessInputPeakTolerance();
+  processInputFitRanges();
+
+  return;
+}
+
+//----------------------------------------------------------------------------------------------
+/** process inputs for peak profile and background
+ * @brief FitPeaks::processInputFunctions
+ */
+void FitPeaks::processInputFunctions() {
+  // peak functions
+  std::string peakfunctiontype = getPropertyValue("PeakFunction");
+  m_peakFunction = boost::dynamic_pointer_cast<IPeakFunction>(
+      API::FunctionFactory::Instance().createFunction(peakfunctiontype));
+
+  // background functions
+  std::string bkgdfunctiontype = getPropertyValue("BackgroundType");
+  std::string bkgdname;
+  if (bkgdfunctiontype.compare("Linear") == 0)
```

Comment:
`FitPeaks.cpp:369:7: warning: do not use 'compare' to test equality of strings; use the string equality operator instead [misc-string-compare]`